core/src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -1492,7 +1492,7 @@ public void readExternal(DataInputStream dis, PrototypeFactory pf) throws IOExce
         extensions = (Vector)ExtUtil.read(dis, new ExtWrapListPoly(), pf);
 
         setEvaluationContext(new EvaluationContext(null));
-        actionController = (ActionController)ExtUtil.read(dis, new ExtWrapNullable(ActionController.class), pf);
+        actionController = (ActionController)ExtUtil.read(dis, new ExtWrapTagged(), pf);
```

Comment:
I'm pretty sure `ExtWrapTagged` is the right way to do this, but not totally sure...